keymanapp/keyman
# feat(windows): UI layout for update tab 💽

Convert the update tab table to use css grid instead of html tables. 
The apply update tab button is greyed out unless there is a Keyboard or Keyman update available.

The next stage of #10045
Fixes: #10045

# User Testing


## TEST_UI_UPDATES_KBD

1. Install the Keyman Attached to the PR.

2. Download the version of sil_eurolatin attached to his PR.

3. Double-click on the downloaded file in Windows Explorer to install the keyboard

4. Open Keyman configuration

5. Ensure the Keyboard is installed and check the version is xx

6. Select the Update Tab on the left.

7. Press the Check for Updates button

8. The window form shall be updated with the new version of the keyboard available.




## TEST_UI_UPDATES_KEYMAN

This test is a little tricky and needs a newer alpha build then this PR.
Note to self: I could force a new build with dummy PR to bump to a newer version than this which will allow for testing.

1. Install the Keyman Attached to the PR.

2. Open Keyman configuration

3. In the tab check the Keyman version

4. Select the Update Tab on the left.

5. Press the Check for Updates button

8. The window form shall be updated with the new version of Keyman Available.

## TEST_DELETE_KEYBOARD_CHECKUPDATES
Test configuration doesn't crash after the following sequence. 
Steps to reproduce:

1. Install Keyman attached the "18.0.153-alpha-test-12840" PR build.
2. Installed the Gff_Amharic 3.1.1 version keyboard.
3. Open the Keyman Configuration dialog.
4. Navigate to the "Keyboard Layouts tab".
5. Verified that the installed keyboard shows an older version.
6. Navigate to the "Update" tab.
7. Press the "Check for new updates" button.
8. Verified that the newer version appeared in the "Updated component" column.
9. Navigate to the "Keyboard Layouts" tab and uninstall the GFF Amharic keyboard.
10. Keyboard should uninstall and Configuration display is refreshed.
